include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,818 @@
+   */
+  MDSPAN_TEMPLATE_REQUIRES(
+      class _OtherExtents,
+      /* requires */ (
+          std::is_constructible_v<extents_type, _OtherExtents>
+          )
+      )
+  MDSPAN_CONDITIONAL_EXPLICIT((extents_type::rank() > 0))
+  constexpr mapping(const layout_stride::mapping<_OtherExtents> &__other_mapping)
+      : __padded_stride(__other_mapping, std::integral_constant<size_t, __padding_stride_idx>{}),
+        __extents(__other_mapping.extents())
+  {}
+
+  /**
+   * Converting constructor from `layout_right_padded::mapping`.
+   *
+   * This overload participates in overload resolution only if `is_constructible_v<extents_type, OtherExtents>` is true.
+   * Either `padding_stride` or `OtherPaddingStride` must be `std::dynamic_extent`, or `padding_stride == OtherPaddingStride`.
+   */
+  MDSPAN_TEMPLATE_REQUIRES(
+      class _Mapping,
+      /* requires */ (
+          detail::__is_layout_right_padded_mapping<_Mapping>::value
+              && std::is_constructible_v<extents_type, typename _Mapping::extents_type>
+          )
+      )
+  MDSPAN_CONDITIONAL_EXPLICIT((extents_type::rank() > 1 &&
+                               (padding_stride == dynamic_extent ||
+                                _Mapping::padding_stride == dynamic_extent)))
+  constexpr mapping(const _Mapping &__other_mapping)
+      : __padded_stride(__other_mapping, std::integral_constant<size_t, __padding_stride_idx>{}),
+        __extents(__other_mapping.extents())
+  {
+    static_assert(padding_stride == dynamic_extent ||
+                  _Mapping::padding_stride == dynamic_extent ||
+                  padding_stride == _Mapping::padding_stride);
+  }
+
+  /**
+   * Converting constructor from `layout_left_padded::mapping`.
+   *
+   * This overload participates in overload resolution only if `extents_type::rank()` is 0 or 1 and `is_constructible_v<extents_type, OtherExtents>` is `true`.
+   */
+  MDSPAN_TEMPLATE_REQUIRES(
+      class _Mapping,
+      /* requires */ (
+          detail::__is_layout_left_padded_mapping<_Mapping>::value
+                  && extents_type::rank() <= 1
+          && std::is_constructible_v<extents_type, typename _Mapping::extents_type>
+          )
+      )
+  MDSPAN_CONDITIONAL_EXPLICIT((!std::is_convertible_v<typename _Mapping::extents_type, extents_type>))
+  constexpr mapping(const _Mapping &__other_mapping) noexcept
+      : __padded_stride(__other_mapping.extents(), __other_mapping.extents().extent(__extent_to_pad_idx)),
+        __extents(__other_mapping.extents())
+  {}
+
+  constexpr const extents_type &extents() const noexcept
+  {
+    return __extents;
+  }
+
+  constexpr std::array<index_type, extents_type::rank()>
+  strides() const noexcept
+  {
+    if constexpr ( extents_type::rank() == 0 ) {
+      return {};
+    } else {
+      index_type value = 1;
+      std::array<index_type, extents_type::rank()> __s{};
+      for (rank_type __r = extents_type::rank() - 1; __r > __extent_to_pad_idx; --__r)
```

Comment:
this will go horribly wrong for rank 1 or? Because __extent_to_pad_idx is something  like numeric_limits<size_t>::max - 1?

Reply:
not here because __r will be 0 for rank 1 and thus not greater than max. But anyway this is going away because of the padded stride not being valid anymore for ranks < 2